### PR TITLE
Change default plugin timeout to 10 seconds instead of 0

### LINKED ISSUE #1090: Make plugintTimeout default be X seconds in v2
as titled

### PR DESCRIPTION
**Pull request by cemremengu**
Fixes #1090 hopefully right this time 😄 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
this should stay as milliseconds.

File: docs/Server.md [hunk 1]
```diff
@@ -189,11 +189,11 @@ fastify.get('/', (request, reply) => {
 <a name="plugin-timeout"></a>
 ### `pluginTimeout`
 
-The maximum amount of time in milliseconds in which a plugin can load.
+The maximum amount of time in *milliseconds* in which a plugin can load.
 If not, [`ready`](https://github.com/fastify/fastify/blob/master/docs/Server.md#ready)
 will complete with an `Error` with code `'ERR_AVVIO_PLUGIN_TIMEOUT'`.
 
-+ Default: `0` (disabled)
++ Default: `10 seconds`
```
Comment (mcollina) [inline docs/Server.md]: should be 10000

**Comment by mcollina** [comment]
I would love to add a test for this. You can use https://www.npmjs.com/package/lolex to mock `setTimeout`.

File: test/plugin.test.js [hunk 2]
```diff
@@ -503,3 +503,19 @@ test('pluginTimeout', t => {
     t.equal(err.code, 'ERR_AVVIO_PLUGIN_TIMEOUT')
   })
 })
+
+test('pluginTimeout default', t => {
+  t.plan(2)
+  const clock = require('lolex').install()
+
```
Comment (mcollina) [inline test/plugin.test.js]: can you move the require to the top, and add an uninstall() in a t.tearDown()?

**Comment by delvedor** [comment]
@mcollina the use of `lolex` could cause some issues with our parallel tests?

**Pull request merged**